cql3/functions/aggregate_fcts.cc

```diff
@@ -497,6 +498,7 @@ bool user_aggregate::is_pure() const { return _sfunc->is_pure() && (!_finalfunc
 bool user_aggregate::is_native() const { return false; }
 bool user_aggregate::is_aggregate() const { return true; }
 bool user_aggregate::requires_thread() const { return _sfunc->requires_thread() || (_finalfunc && _finalfunc->requires_thread()); }
+void user_aggregate::set_reducefunc(::shared_ptr<scalar_function> reducefunc) { _reducefunc = reducefunc; }
```

Comment:
Why do we need this in addition to the constructor? It's not bad per se, but in general function objects should be immutable so we don't have to worry about updates and invalidation.